### ISSUE TITLE
multigit --status fails on up-to-date repositories

### ISSUE THREAD

**Issue by jmnavarrol** [jmnavarrol/python-multigit#7]
Running `multigit --status` on some status throws an uncontrolled exception.
```sh
$ multigit --status
INFO: processing 'subrepos' file rooted at '/home/jmnav/REPOS/python-multigit':
'projects/simplest-git-subrepos/'
        repository: 'git@github.com:jmnavarrol/simplest-git-subrepos.git'
        requested branch: 'python-example'
Traceback (most recent call last):
  File "/home/jmnav/.virtualenvs/multigit-development/bin/multigit", line 8, in <module>
    sys.exit(main())
  File "/home/jmnav/REPOS/python-multigit/src/multigit/__init__.py", line 50, in main
    my_subrepos.process(os.getcwd(), report_only=args.status)
  File "/home/jmnav/REPOS/python-multigit/src/multigit/subrepos.py", line 121, in process
    self.__print_subrepo_status(current_subrepo)
  File "/home/jmnav/REPOS/python-multigit/src/multigit/subrepos.py", line 363, in __print_subrepo_status
    if subrepo['status'] == 'ERROR':
KeyError: 'status'
$
```
This bug was introduced on **v1.10.5**.

**Comment by jmnavarrol** [comment]
This bug highlights an underlying problem: method  Subrepos.__process_subrepo() is already a too big ball of hair, so it's time to refactor it.

Most possibly it'll go in the form of a new Gitrepo class tasked with the management of a single repository.

In the meantime, it's probably more stable downgrading to v0.10.2.

**Comment by jmnavarrol** [comment]
Still in development, I think version **[0.11.0.dev1](https://pypi.org/project/multigit/0.11.0.dev1/)** is already in better shape than 0.10.x series, so I published it.

At the very least, 0.11.0.dev1 doesn't show this current bug.

**Comment by jmnavarrol** [comment]
New version published: [**v0.11.0**](https://pypi.org/project/multigit/0.11.0/).

This should fix this bug and make development easier in the future so further bugs of this nature are hopefully avoided.
